cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/PeriodicReplicationService.java

```diff
@@ -66,11 +66,17 @@
     private static final String PERIODIC_REPLICATION_ENABLED_SUFFIX
         = ".periodicReplicationsActive";
 
+    /* We store a flag indicating whether periodic replications were explicitly stopped in
+     * SharedPreferences using a key with this suffix. We have to store the flag persistently as
+     * the service may be stopped and started by the operating system. */
+    private static final String EXPLICITLY_STOPPED_SUFFIX = ".explicitlyStopped";
+
     private static final long MILLISECONDS_IN_SECOND = 1000L;
 
     public static final int COMMAND_START_PERIODIC_REPLICATION = 2;
     public static final int COMMAND_STOP_PERIODIC_REPLICATION = 3;
     public static final int COMMAND_DEVICE_REBOOTED = 4;
+    public static final int COMMAND_RESET_REPLICATION_TIMERS = 5;
```

Comment:
Added comment in 3751a3a.  The value 5 is chosen because it's one more than the current highest of the `COMMAND_*` values currently in use. See my additional comment also about this, which I've made separately so it doesn't get folded by GitHub.